src/cmd_line/commands/file.ts

```diff
@@ -32,109 +32,94 @@ export class FileCommand extends node.CommandBase {
-        await vscode.commands.executeCommand('workbench.action.closeOtherEditors');
+        await hidePreviousEditor();
       }
       return;
     }
 
+    let filePath = '';
+
+    // Using the empty string will request to open a file
     if (this._arguments.name === '') {
-      if (this._arguments.position === FilePosition.NewWindow) {
-        await vscode.commands.executeCommand('workbench.action.splitEditor');
+      // No name on split is fine and just return
+      if (split === true) {
         return;
       }
 
       const fileList = await vscode.window.showOpenDialog({});
       if (fileList) {
-        const doc = await vscode.workspace.openTextDocument(fileList[0]);
-        vscode.window.showTextDocument(doc);
+        filePath = fileList[0].fsPath;
       }
-      return;
-    }
-
-    this._arguments.name = <string>untildify(this._arguments.name);
-
-    let filePath = this._arguments.name;
-    if (!path.isAbsolute(this._arguments.name)) {
-      let curFilePath = vscode.window.activeTextEditor!.document.uri.fsPath;
-      filePath = path.join(path.dirname(curFilePath), this._arguments.name);
+    } else {
+      // Using a filename, open or create the file
+      this._arguments.name = <string>untildify(this._arguments.name);
+      filePath = path.isAbsolute(this._arguments.name)
+        ? this._arguments.name
+        : path.join(path.dirname(editorFilePath), this._arguments.name);
 
-      if (!fs.existsSync(filePath)) {
+      if (filePath !== editorFilePath && !fs.existsSync(filePath)) {
         // if file does not exist and does not have an extension
         // try to find it with the same extension
         if (path.extname(filePath) === '') {
-          const pathWithExt = filePath + path.extname(curFilePath);
+          const pathWithExt = filePath + path.extname(editorFilePath);
           if (fs.existsSync(pathWithExt)) {
             filePath = pathWithExt;
           }
         }
+        if (this._arguments.new) {
+          fs.closeSync(fs.openSync(filePath, 'w'));
+        } else {
+          Message.ShowError('This file ' + filePath + ' does not exist.');
+          return;
+        }
       }
     }
 
-    // create file
-    if (this.arguments.createFileIfNotExists) {
-      fs.closeSync(fs.openSync(filePath, 'w'));
-    } else {
-      Message.ShowError('The file ' + filePath + ' does not exist.');
-      return;
-    }
-
-    let folder = vscode.Uri.file(filePath);
-    await vscode.commands.executeCommand(
-      'vscode.open',
-      folder,
-      this._arguments.position === FilePosition.NewWindow
-        ? this.getViewColumnToRight()
-        : this.getActiveViewColumn()
-    );
-
-    if (this.arguments.lineNumber) {
-      vscode.window.activeTextEditor!.revealRange(
-        new vscode.Range(
-          new vscode.Position(this.arguments.lineNumber, 0),
-          new vscode.Position(this.arguments.lineNumber, 0)
-        )
-      );
+    if (filePath !== '') {
```

Comment:
I don't think there's a case where this would be false?

Reply:
You are right. I have removed the check.